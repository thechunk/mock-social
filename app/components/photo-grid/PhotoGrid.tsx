import * as React from 'react';
import {PureComponent} from "react";
import {FlatList, Dimensions, PixelRatio} from "react-native";
import withUserPhotosData from "../../hoc/withAlbumPhotosData";
import PhotoGridCell from "./PhotoGridCell";
import {NavigationInjectedProps} from "react-navigation";
import withNavigationParamsAsProps from "../../hoc/withNavigationParamsAsProps";

class PhotoGrid extends PureComponent<IPhotoGrid & NavigationInjectedProps & IWithUserPhotosDataProps> {
    private columns: number = 5;
    private imageWidth: number = 0;

    componentDidMount(): void {
        const {width} = Dimensions.get('window');
        this.imageWidth = Math.floor(width / this.columns);
    }

    render() {
        return (
            <FlatList
                data={this.props.photos}
                numColumns={this.columns}
                keyExtractor={v => v.id.toString()}
                renderItem={v => (
                <PhotoGridCell
                    url={v.item.url}
                    style={{width: this.imageWidth, height: this.imageWidth}} />
            )} />
        );
    }
}

export default withNavigationParamsAsProps(withUserPhotosData(PhotoGrid));