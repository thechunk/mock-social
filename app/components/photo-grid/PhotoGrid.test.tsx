import * as React from 'react';
import {shallow} from "enzyme";
import {PhotoGrid} from "./PhotoGrid";

describe('PhotoGrid', () => {
    it('renders correctly', () => {
        expect(shallow(<PhotoGrid/>)).toMatchSnapshot();
    });
});