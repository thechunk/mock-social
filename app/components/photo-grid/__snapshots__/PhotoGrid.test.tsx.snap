// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGrid renders correctly 1`] = `
<FlatList
  contentContainerStyle={
    Object {
      "paddingBottom": 25,
      "paddingTop": 89,
    }
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={3}
  numColumns={4}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  scrollIndicatorInsets={
    Object {
      "bottom": 25,
      "top": 89,
    }
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
